modin/numpy/arr.py

```diff
@@ -1467,41 +1562,308 @@ def sum(
                     )
-                if where is not False or out is not None:
+                if truthy_where or out is not None:
                     return fix_dtypes_and_determine_return(
                         array(numpy.array([[result]], dtype=out_dtype))._query_compiler,
                         2,
                         dtype,
                         out,
-                        where is not False,
+                        truthy_where,
                     )
                 else:
                     return array([[initial]], dtype=out_dtype)
-            return result if where is not False else initial
+            return result if truthy_where else initial
         target = where.where(self, 0) if isinstance(where, array) else self
         result = target._query_compiler.astype(
             {col_name: out_dtype for col_name in target._query_compiler.columns}
         ).sum(axis=axis, skipna=False)
         result = result.add(initial)
         new_ndim = self._ndim - 1 if not keepdims else self._ndim
         if new_ndim == 0:
-            return result.to_numpy()[0, 0] if where is not False else initial
+            return result.to_numpy()[0, 0] if truthy_where else initial
         if not keepdims and axis != 1:
             result = result.transpose()
         if out is not None:
             out._query_compiler = (
                 (numpy.ones_like(out) * initial).astype(out_dtype)._query_compiler
             )
-        if where is not False or out is not None:
+        if truthy_where or out is not None:
             return fix_dtypes_and_determine_return(
-                result, new_ndim, dtype, out, where is not False
+                result, new_ndim, dtype, out, truthy_where
             )
         else:
             return (
                 numpy.zeros_like(array(_query_compiler=result, _ndim=new_ndim))
                 + initial
             )
 
+    def all(self, axis=None, out=None, keepdims=None, *, where=True):
+        check_kwargs(keepdims=keepdims, where=where)
+        truthy_where = bool(where)
+        target = where.where(self, True) if isinstance(where, array) else self
+        if self._ndim == 1:
+            if axis == 1:
+                raise numpy.AxisError(1, 1)
+            result = target._query_compiler.all(axis=0)
+            if keepdims:
+                if out is not None and out.shape != (1,):
+                    raise ValueError(
+                        f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
+                    )
+                if truthy_where or out is not None:
+                    return fix_dtypes_and_determine_return(
+                        result, 1, bool, out, truthy_where
+                    )
+                else:
+                    return array([True], dtype=bool)
+            result = result.to_numpy()[0, 0] if truthy_where else True
+            return result if where else True
```

Comment:
This should be `truthy_where`.

Reply:
but couldn't we also just return result in this line? Since we've already made sure it has the correct value in line 1619?